Sources/Classes/Headers/Public/RSClient.h

```diff
@@ -89,7 +89,8 @@ typedef void (^Callback)(NSObject *_Nullable);
 + (RSOption*) getDefaultOptions __attribute((deprecated("This method will be deprecated soon. Use instance property(defaultOptions) instead.")));
 - (RSContext *) getContext __attribute((deprecated("This method will be deprecated soon. Use instance property(context) instead.")));
 
-- (void) onIntegrationReady:(NSString *)integrationName withCallback:(Callback)callback;
+- (void) onIntegrationReadyWithString:(NSString *)integrationName withCallback:(Callback)callback;
```

Comment:
Can we just keep it `onIntegrationReady` ?

Reply:
Method overloading doesn't seem to be supported by Objective-C. I initially tried that way by changing the argument type but it gave a warning/error. Hence, I choose this way. Also, refer here: https://stackoverflow.com/questions/2286312/method-overloading-in-objective-c.

Reply:
I believe we do not need method overloading in this case. We can keep the method names as below:
i) onIntegrationReady
ii) onIntegrationReadyWithFactory